### PR TITLE
[sqlite3] Add feature recovery

### LINKED ISSUE #39396: [sqlite3] Lack of DB recovery feature.
### Is your feature request related to a problem? Please describe.

I would do recover a corrupt database, without this feature I could not do that.
_sqlite3 corrupted-db.db ".recover" | sqlite3 recovered-db.db_

### Proposed solution

Please support this feature and provide an optional argument for it, some thing like "recovery" as below
_vcpkg install sqlite3[core,recovery]:x64-windows_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/39396

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

This feature passed with following triplet:

```
x64-windows
```

### CODE REVIEW

**Review by WangWeiLin-MV — APPROVED** [review]

**Review by kadirlua — APPROVED** [review]

File: ports/sqlite3/portfile.cmake [hunk 1]
```diff
@@ -41,6 +41,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS none # only using the script-mode side-
         memsys3             SQLITE_ENABLE_MEMSYS3
         memsys5             SQLITE_ENABLE_MEMSYS5
         limit               SQLITE_ENABLE_UPDATE_DELETE_LIMIT
+        recovery            SQLITE_ENABLE_DBPAGE_VTAB
```
Comment (dg0yt) [inline ports/sqlite3/portfile.cmake:44]: I do understand that `SQLITE_ENABLE_DBPAGE_VTAB` is necessary to provide "recovery".  
But OTOH it is not obvious that I need to activate `recovery` to get ["The SQLITE_DBPAGE Virtual Table"](https://sqlite.org/dbpage.html).

Should there be two separate features, with `recovery` depending on `dbpage-vtab`?

Comment (jimwang118) [reply ports/sqlite3/portfile.cmake:44]: I don't understand what you mean. Are you saying that [recovery](https://sqlite.org/recovery.html) functionality depends on other compilation options?

**Review by WangWeiLin-MV — DISMISSED** [review]
The port installation tests pass with the following triplets:
* x64-linux

File: ports/sqlite3/portfile.cmake (hunk 1 above)
Comment (BillyONeal) [reply ports/sqlite3/portfile.cmake:44]: I think @dg0yt means "it is surprising that the feature that sets SQLITE_ENABLE_DBPAGE_VTAB, and thus enables the 'SQLITE_DBPAGE virtual table' as documented at https://sqlite.org/compile.html#_options_to_enable_features_normally_turned_off , is named 'recovery'". Perhaps it should be named `dbpage-vtab` instead.

Notably, sqlite's documentation ( https://sqlite.org/recovery.html#how_to_implement_recovery ) says "must be compiled with the following option: `-DSQLITE_ENABLE_DBPAGE_VTAB`" so folks following what the documentation says might be looking for `dbpage-vtab`.

Would you object to giving the feature that name?

Comment (jimwang118) [reply ports/sqlite3/portfile.cmake]: Done

**Review by WangWeiLin-MV — APPROVED** [review]

File: ports/sqlite3/vcpkg.json [hunk 2]
```diff
@@ -19,6 +19,9 @@
     "json1"
   ],
   "features": {
+    "dbpage-vtab": {
+      "description": "Enable the recovery extension"
```
Comment (dg0yt) [inline ports/sqlite3/vcpkg.json:23]: Looks like the description must be fixed in another PR. :sob:

Comment (jimwang118) [reply ports/sqlite3/vcpkg.json:23]: > Looks like the description must be fixed in another PR. 😭

Does the description need to be modified to enable the SQLITE_DBPAGE virtual table?

Comment (dg0yt) [reply ports/sqlite3/vcpkg.json:23]: No, it needs to be modified for vcpkg user experience.  
"I looked for a feature to enable sqlite's SQLITE_DBPAGE extension, but I couldn't find it!"
